SIPS/sip-15.md

```diff
@@ -0,0 +1,73 @@
+---
+sip: 15
+title: Snaps Home Page
+status: Draft
+discussions-to: https://github.com/MetaMask/SIPs/discussions/115
+author: Frederik Bolding (@frederikbolding)
+created: 2023-10-26
+---
+
+## Abstract
+
+This SIP proposes a new API that allows snaps to surface a static UI to display data that may be useful to users of the snap. This proposal outlines some of the details around this feature.
+
+## Motivation
+
+Snaps currently tend to leverage separate websites for showing details relating to the status of the snap. For example, snaps that manage new blockchain accounts for users have to rely almost exclusively on these to show balances on new kinds of networks. This proposal aims to start improving on this by giving snaps a surface directly in the client where they can render custom UI. 
+
+This new surface differs from the existing custom UI surfaces as it isn't reactionary, instead of being triggered by an RPC call or a transaction being confirmed the user can choose to view the home page screen at any time.
+
+## Specification
+
+> Formal specifications are written in Typescript.
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+"OPTIONAL" written in uppercase in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### Snap Manifest
+
+This SIP specifies a permission named `endowment:home-page`.
```

Comment:
Let's say we eventually have different pages, would it be better to have: 

- `endowment:page-home`
- `endowment:page-tx-history`
- `endowment:page-send`

Instead of the pattern being : [location] - page?

Reply:
I don't think it changes much, but we can make the change if you feel that it improves the pattern

Reply:
I can see the above pages being useful for a protocol snap to integrate into our UI, but at that point we're permitting divergent ux within the extension and also contradicting our position of non-reactionary UI. I think having a home-page as a catch-all is a good intermediary step to direct integration to different parts of the extension i.e. tx-history, account balances etc.

Reply:
These are bad examples but just assuming that there would eventually be different types of pages, maybe the pattern `page-[location]` is better than `[location]-page`